catena4450m102_waterlevel/catena4450m102_waterlevel.ino

```diff
@@ -0,0 +1,697 @@
+        }
+
+  if (fBme)
+       {
+       Adafruit_BME280::Measurements m = bme.readTemperaturePressureHumidity();
+       // temperature is 2 bytes from -0x80.00 to +0x7F.FF degrees C
+       // pressure is 2 bytes, hPa * 10.
+       // humidity is one byte, where 0 == 0/256 and 0xFF == 255/256.
+       gCatena.SafePrintf(
+                "BME280:  T: %d P: %d RH: %d\n",
+                (int) m.Temperature,
+                (int) m.Pressure,
+                (int) m.Humidity
+                );
+       b.putT(m.Temperature);
+       b.putP(m.Pressure);
+       b.putRH(m.Humidity);
+
+       flag |= FlagsSensor4::FlagTPH;
+       }
+
+  if (fLux)
+        {
+        /* Get a new sensor event */
+        uint16_t light;
+
+        light = bh1750.readLightLevel();
+        gCatena.SafePrintf("BH1750:  %u lux\n", light);
+        b.putLux(light);
+        flag |= FlagsSensor4::FlagLux;
+        }
+		
+  /*
+  || Measure and transmit the RC601 water pressure transducer values.
+  || No seperate library cerated. Used the I2C wire library controls.
+  || Calculation formula of current pressure value of transmitter:
+  || current pressure value = （current read value - reference value）/（cross value）*（transmitter range）
+  || The reference value is fixed to 1677721; The cross value is fixed to 13421772;
+  || The calculation of the output pressure value of 1000KPa transmitter is
+  || as follows:
+  || If the current read data is 3100429,
+  || Calculation：（3100429-1677721）/（13421772）*（1000KPa）= 106KPa
+  */
+  if (AuroPressureSensor)
+		{
+		unsigned char pBuffer[10]={0};
+		uint8_t Index, nBytes;
+		uint32_t SenAD;
+		float SenPDat;
+		
+		/* Initialize the I2C MUX and Choose the second channel*/
+		Wire.beginTransmission(0x70);
+		Wire.write(0x05); 
+		Wire.endTransmission();
+		delay(1000);
+		
+		/* Sending acquisition instruction to Sensor*/
+		Wire.beginTransmission(0); 
+		Wire.write(0xaa);      
+		Wire.write(0x00);      
+		Wire.write(0x00);     
+		Wire.endTransmission();      
+		delay(100);                  
+		
+		/* 
+		Reading the output data frame contains: status, pressure and temperature
+		data; total 8 bytes; The first byte of the output frame is a status
+		byte; The 2~4 byte is 24 bit unsigned pressure data, high byte in front and
+		low byte in back; The 5~7 byte is 24 bit unsigned temperature data, high byte
+		in front, low byte in back 
+		*/
+		nBytes = 7;
+		Wire.requestFrom(0, nBytes);    
+
+		if (nBytes <= Wire.available()) 
+			{ 
+			/* Reading the nBytes of Data and storing it internal buffer*/
+			Index = 0;
+			while (Index < nBytes) 
+				{
+				pBuffer[Index++] = Wire.read();
+				}   
+			}
+		
+		SenAD = (uint32_t) ((pBuffer[1] << 16) + (pBuffer[2] << 8) + pBuffer[3]);
+		gCatena.SafePrintf("SenAD: %u\n", SenAD);
+		
+		if (SenAD >= REFERENCE_VALUE) /* Check for negative value, If yes then print 0*/
+			{
+			SenPDat = (float)(SenAD - REFERENCE_VALUE) / CROSS_VALUE * RANGE;
+			Serial.print("SenPDat: ");
```

Comment:
This will not work if there's no USB attached. Use safePrintf or similar.

Reply:
Ok sure, Will use safePrintf